mc/query/query.peg

```diff
@@ -0,0 +1,91 @@
+# Grammar for statement query language
+# just simple statements with no joins for now
+#
+# Compilation to query.peg.go:
+#  go get github.com/pointlander/peg
+#  peg -inline -switch query.peg
+
+package query
+
+type QueryParser Peg {
+    *ParseState
+}
+
+Grammar <- 'SELECT' WS Selector
+                    WS Source
+                    (WS Criteria)?
+                    (WS Limit)?
+                    WSX EOF
+
+Selector <- SimpleSelector   { p.setSimpleSelector() }
+          / CompoundSelector { p.setCompoundSelector() }
+          / FunctionSelector { p.setFunctionSelector() }
+
+SimpleSelector <- < SimpleSelectorOp > { p.push(text) }
+SimpleSelectorOp <- '*'
+                  / 'body'
+                  / 'id'
+                  / 'publisher'
+                  / 'namespace'
+                  / 'source'
+                  / 'timestamp'               
+
+CompoundSelector <- '(' SimpleSelector ( ',' WSX SimpleSelector )* ')'
+
+FunctionSelector <- Function '(' SimpleSelector ')'
+
+Function   <- < FunctionOp > { p.push(text) }
+FunctionOp <- 'COUNT'
```

Comment:
No.
The function is applied in the result set generated by the selector. So different selector will get different counts, because of set semantics in the slection.